homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -191,36 +191,39 @@ def state_received(msg):
                 self._state = False
 
             if self._supported_features and SUPPORT_COLOR:
-                try:
-                    red = int(values["color"]["r"])
-                    green = int(values["color"]["g"])
-                    blue = int(values["color"]["b"])
-
-                    self._hs = color_util.color_RGB_to_hs(red, green, blue)
-                except KeyError:
-                    pass
-                except ValueError:
-                    _LOGGER.warning("Invalid RGB color value received")
-
-                try:
-                    x_color = float(values["color"]["x"])
-                    y_color = float(values["color"]["y"])
-
-                    self._hs = color_util.color_xy_to_hs(x_color, y_color)
-                except KeyError:
-                    pass
-                except ValueError:
-                    _LOGGER.warning("Invalid XY color value received")
-
-                try:
-                    hue = float(values["color"]["h"])
-                    saturation = float(values["color"]["s"])
-
-                    self._hs = (hue, saturation)
-                except KeyError:
-                    pass
-                except ValueError:
-                    _LOGGER.warning("Invalid HS color value received")
+                if "color" in values and values["color"] is None:
```

Comment:
Does it make sense to make this its own function that returns the `hs` as soon as it has a valid value?

Reply:
Yeah, fixed.